python/chronos/src/bigdl/chronos/data/experimental/xshards_tsdataset.py

```diff
@@ -364,6 +364,33 @@ def _inverse_transform(df, id_col, scaler, feature_col, target_col):
                                                   self.target_col)
         return self
 
+    def unscale_xshards(self, data):
+        '''
+        Unscale the time series forecaster's numpy prediction result/ground truth.
+
+        :param data: xshards same with self.numpy_xshards.
+
+        :return: the unscaled xshardtsdataset instance.
+        '''
+        def _inverse_transform(data, scaler, scaler_index):
+            from sklearn.utils.validation import check_is_fitted
+            from bigdl.nano.utils.log4Error import invalidInputError
+
+            id = data['id'][0, 0]
+            scaler_for_this_id = scaler[id]
+            invalidInputError(not check_is_fitted(scaler_for_this_id),
+                              "scaler is not fitted. When calling scale for the first time, "
+                              "you need to set fit=True.")
+
+            return unscale_timeseries_numpy(data['y'], scaler_for_this_id, scaler_index)
+
+        def _get_features(df):
+            return df.columns
+
+        cols = self.shards.transform_shard(_get_features).collect()[0]
+        scaler_index = [cols.get_loc(col) + 1 for col in self.target_col]
```

Comment:
I think +1 is kind of like a ad-hoc patch for your specific test data. The real problem is:
1. You can never know if target_col(e.g. "value") is ahead of feature_col(e.g. "extra_features") or not
2. The order in your scaler is [feature_col, target_col]
3. That's why the order is wrong!


Solution is:
1. change the order in your scaler to [target_col, feature_col] (just like what we do in TSDataset)
2. get the scaler_index in roll rather than unscale (just like what we do in TSDataset)
3. scaler_index can be set to `[self.target_col.index(feature_col[i]) for i in range(len(feature_col))] ` (just like what we do in TSDataset)